RIOT-OS/RIOT
# Rust: Update to riot-wrappers 0.9

### Contribution description

This updates riot-wrappers from the 0.8 series to 0.9. The [change log](https://github.com/RIOT-OS/rust-riot-wrappers/blob/main/NEWS.md) lists what happened: Most are changes that were in the pipeline for quite some time and waited for a critical mass to do a breaking change. Being breaking, this also required changes both to the provided examples and to the example modules that are referenced -- this update rolls them all into one PR.

### Testing procedure

Automated tests should suffice. Run `make -C examples/rust-gcoap` for good measure.

### Issues/PRs references

Some related PRs are listed at https://github.com/RIOT-OS/rust-riot-wrappers/issues?q=label%3Await-for-breaking-0.9; the big 0.9 transition is also described in https://github.com/RIOT-OS/rust-riot-wrappers/pull/109
